### PR TITLE
async adapter bug fixes; observe and observeQuery bug fixes; empty key value bug fix

### PR DESCRIPTION
**Pull request by david-mcafee**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/datastore/src/datastore/datastore.ts [hunk 1]
```diff
@@ -1347,6 +1357,13 @@ class DataStore {
 					criteria
 				);
 			} else {
+				// observeQuery does not accept OL, okay to remove, Manuel?
```
Comment (david-mcafee) [inline packages/datastore/src/datastore/datastore.ts:1360]: Discussed with Manuel offline, will remove

File: packages/datastore/src/util.ts [hunk 2]
```diff
@@ -31,8 +31,13 @@ import {
 	IndexesType,
 } from './types';
 import { WordArray } from 'amazon-cognito-identity-js';
+
 const ID = 'id';
 
+export const errorMessages = {
+	idEmptyString: 'Field of type `ID` cannot be an empty string',
```
Comment (iartemiev) [inline packages/datastore/src/util.ts:38]: This should read: "An index field cannot contain an empty string value". 

The field's type is not important, we can have an optional field with type `ID`. But any field (regardless of type) that's part of a primary key or other index cannot be empty.

**Review by iartemiev — CHANGES_REQUESTED** [review]
Looks good! Just a couple tiny call outs

File: packages/datastore/src/util.ts (hunk 2 above)
Comment (david-mcafee) [reply packages/datastore/src/util.ts:38]: Updated!

**Review by iartemiev — APPROVED** [review]
LGTM! 👍🏻

**Review by dpilch — APPROVED** [review]
LGTM, just one non-blocking style nit

File: packages/datastore/src/datastore/datastore.ts [hunk 3]
```diff
@@ -987,6 +989,14 @@ class DataStore {
 
 		const modelDefinition = getModelDefinition(modelConstructor);
 
+		for (const key of extractPrimaryKeyFieldNames(modelDefinition)) {
```
[the remark this answers is not in the document]
Comment (dpilch) [reply packages/datastore/src/datastore/datastore.ts]: nvm, looks like this has changed since I started the review.

**Pull request merged**